src/core/services/aria/aria.js

```diff
@@ -62,7 +62,26 @@ function AriaService($$rAF, $log, $window, $interpolate) {
   }
 
   function getText(element) {
-    return (element.text() || "").trim();
+    element = element[0] || element;
+    var walker = document.createTreeWalker(element, NodeFilter.SHOW_TEXT, null, false);
+    var text = '';
+
+    var node;
+    while (node = walker.nextNode()) {
```

Comment:
Rather than walking through all of the nodes here, can you not just use the third parameter `filter` of the `createTreeWalker()` method? I would assume you could just pass the `isAriaHiddenNoe()` method to it (wrapped in an object I think) and it would filter the nodes for you.

I guess you would still have to traverse the remaining nodes to add the text content, but it might shorten the code for you a bit.